src/settings/accounts-settings.test.tsx

```diff
@@ -0,0 +1,756 @@
+		const persona3Row = screen.getByText('New Persona 3');
+		expect(persona3Row).toBeVisible();
+
+		await user.click(persona1Row);
+		await user.click(screen.getByRole('button', { name: /delete/i }));
+		let confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
+		await user.click(confirmButton);
+		expect(persona1Row).not.toBeInTheDocument();
+
+		await user.click(persona2Row);
+		await user.click(screen.getByRole('button', { name: /delete/i }));
+		confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
+		await user.click(confirmButton);
+		expect(persona2Row).not.toBeInTheDocument();
+
+		await user.click(screen.getByRole('button', { name: /save/i }));
+
+		const request = await pendingBatchRequest;
+
+		const requestBody = (request?.body as { Body: { BatchRequest: BatchRequest } }).Body;
+		expect(requestBody.BatchRequest.CreateIdentityRequest).toHaveLength(1);
+		expect(requestBody.BatchRequest.DeleteIdentityRequest).toBeUndefined();
+		expect(requestBody.BatchRequest.ModifyIdentityRequest).toBeUndefined();
+
+		const successSnackbar = await screen.findByText('Edits saved correctly');
+		expect(successSnackbar).toBeVisible();
+	});
+
+	test('Should remove from the list added identities not saved on discard changes', async () => {
+		const defaultFirstName = faker.person.firstName();
+		const defaultLastName = faker.person.lastName();
+		const defaultFullName = faker.person.fullName({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultEmail = faker.internet.email({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultId = faker.string.uuid();
+
+		const identitiesArray: Account['identities']['identity'] = [
+			{
+				id: defaultId,
+				name: 'DEFAULT',
+				_attrs: {
+					zimbraPrefIdentityName: defaultFullName,
+					zimbraPrefReplyToEnabled: 'FALSE',
+					zimbraPrefFromDisplay: defaultFirstName,
+					zimbraPrefFromAddress: defaultEmail,
+					zimbraPrefIdentityId: defaultId
+				}
+			}
+		];
+
+		const account: Account = {
+			name: defaultEmail,
+			rights: { targets: [] },
+			signatures: { signature: [] },
+			id: defaultId,
+			displayName: '',
+			identities: {
+				identity: identitiesArray
+			}
+		};
+
+		const identitiesDefault = map(identitiesArray, (item, index) =>
+			identityToIdentityProps(item, index)
+		);
+
+		identitiesDefault.unshift(identitiesDefault.pop() as IdentityProps);
+
+		const { user } = setup(
+			<AccountsSettings account={account} identitiesDefault={identitiesDefault} />
+		);
+		await screen.findByText('sendAs');
+
+		await user.click(screen.getByRole('button', { name: /add persona/i }));
+		const persona1Row = screen.getByText('New Persona 1');
+		expect(persona1Row).toBeVisible();
+
+		await user.click(screen.getByRole('button', { name: /discard changes/i }));
+		expect(persona1Row).not.toBeInTheDocument();
```

Comment:
Also here, persona1 is added and automatically selected, so the delegates section is unmounted. Then on discard, persona1 is removed from the list and default identity is selected again, an the delegates section is rendered again, network request, etc.
```suggestion
        await screen.findByText('sendAs');
		expect(persona1Row).not.toBeInTheDocument();
```